src/server.cpp

```diff
@@ -3707,13 +3707,14 @@ bool Server::dynamicAddMedia(const DynamicMediaArgs &a)
 	// Push file to existing clients
 	if (m_env) {
 		NetworkPacket pkt(TOCLIENT_MEDIA_PUSH, 0);
-		pkt << raw_hash << filename << static_cast<bool>(a.ephemeral);
+		pkt << raw_hash << filename;
 
 		NetworkPacket legacy_pkt = pkt;
 
 		// Newer clients get asked to fetch the file (asynchronous)
-		pkt << a.token;
+		pkt << static_cast<bool>(a.ephemeral) << a.token;
 		// Older clients have an awful hack that just throws the data at them
+		legacy_pkt << static_cast<bool>(!a.ephemeral);	// The client calls this field "cached".
```

Comment:
this is actually not what I had in mind, sorry if that wasn't clear.
the server should always send `!a.ephemeral`.
that way the following combinations will work:
* old client and future server
* current/new client and future server
* (any client and old server)

so only server owners will need to update their servers to fix this.

I'd then put a comment like this:
```c
// NOTE: the meaning of this field was accidentally inverted between proto 39 and 40,
// when dynamic_add_media v2 was added. As of 5.12.0 the server sends it correctly again.
// Compatibility code on the client-side was not added.
```

Reply:
That actually fits in better with networkprotocol.h, which calls this "should_be_cached" (pointed out by Krock on IRC).

Reply:
But in response to (any client and old server) - that will not work.

Old servers will do the "wrong" thing, and new clients will not be any different, so they will do the "wrong" thing.  I have just tested this with patched-client-5.10 to server-5.4, and it does not behave as the documentation says it should.

Which is not a surprise.